### ISSUE TITLE
discussion: migrate to codeberg

### ISSUE THREAD

**Issue by sevz17** [djpohly/dwl#502]
Hi @djpohly, I mentioned it in the discord (btw, can you make me admin of it) but now that you are back it's perfect to discuss about moving the project to codeberg, at first I only wanted to move this repository to an organization, but I thought 'if I'm moving it, why not move it to a hosting not owned by a big tech' and after considering I think codeberg is the best option.

As for the reason why I want to move it, is because I want to start looking for new maintainers but I can't add a notice in the readme without knowing if I can actually add them. (I emailed you in August and I still haven't had any response, added to the fact that you appear only when I create a PR to do something you don't like, stay active a few days and then pass months without knowing about you)

EDIT: I'm not asking you to be active 24/7, I know and understand you have other things to do, but does that really stop you from taking, I don't know 30 minutes, to review open PRs?, it's not like I create 20 pr per day.

EDIT 2: I was planning to reply about the sloc limit today, but I didn't think writing about this would make me so angry, will reply once I calm down.

**Comment by djpohly** [comment]
I like the idea.  I'm definitely down to migrate to a less encumbered, less cumbersome platform, and codeberg looks ideal.

Some backstory: up to this point, I've had a professional reason for dwl to stay in one place, connected with the entire project history.  Creating dwl is part of the "scholarly output" I've been planning to include in my application for tenure, along with the other research and paper-writing I've had to focus on in recent years.  I wanted to make sure the project's significance (stars), review/feedback process (contributors), and longevity (commit history) are clear to see, especially for non-computer-scientists who might not be as familiar with how open-source works.

However... as of this past Friday (!), the official application is submitted.  The decision happens sometime in December, after which I have zero reservations about migrating/restructuring/etc.  We can definitely bring in additional maintainers before that if you have people in mind.  I also went ahead and created a dwl repository on codeberg for trying out in the meantime, and I've added you as an admin.

(Also, I've updated your role on Discord - thanks!)

**Comment by odrling** [comment]
> I also went ahead and created a [dwl repository on codeberg](https://codeberg.org/djpohly/dwl)

It should be possible to migrate all the issues, pull requests and the wiki to the new repository.
Codeberg has this tool available when creating the repo: https://docs.codeberg.org/advanced/migrating-repos/#migrating-from-services

**Comment by sevz17** [comment]
> Some backstory: up to this point, I've had a professional reason for dwl to stay in one place, connected with the entire project history. Creating dwl is part of the "scholarly output" I've been planning to include in my application for tenure, along with the other research and paper-writing I've had to focus on in recent years. I wanted to make sure the project's significance (stars), review/feedback process (contributors), and longevity (commit history) are clear to see, especially for non-computer-scientists who might not be as familiar with how open-source works.

I feel... used.

> However... as of this past Friday (!), the official application is submitted. The decision happens sometime in December, after which I have zero reservations about migrating/restructuring/etc.

Which means...?, use this repo as a mirror?

> > I also went ahead and created a [dwl repository on codeberg](https://codeberg.org/djpohly/dwl)
> 
> It should be possible to migrate all the issues, pull requests and the wiki to the new repository. Codeberg has this tool available when creating the repo: https://docs.codeberg.org/advanced/migrating-repos/#migrating-from-services

Yep, before this I did a "testing" repository: https://codeberg.org/dwl/dwl

**Comment by sevz17** [comment]
@djpohly I feel obligated to ask: Will you disappear **again** for months?

**Comment by dimkr** [comment]
Why not just fork the project on codeberg? Users and patch authors will follow.

**Comment by djpohly** [comment]
@sevz17 I'm sorry I wasn't clearer above - I did not and do not claim credit for anything beyond creating the project and publishing the initial releases. The fact that others have contributed to, built on, and even taken the reins of the project as you have done was actually a very important point I made in my application. It's proper to give credit where it's due, and you are due a lot of credit for your leadership with dwl. (My name hasn't even appeared in the source code or copyrights since 2020, because I wanted it to be clear that this was *not* a solo effort.)

> use this repo as a mirror?

Heck, at that point, I'd even be fine with archiving the GitHub repo and pointing everything to Codeberg. I just can't change the URLs on an application that's already submitted, so there needs to be something here for people to examine in the meantime.

> Yes, before this I did a "testing" repository: https://codeberg.org/dwl/dwl

Excellent, that's even better!
